Formula/c/chapel.rb

```diff
@@ -17,7 +17,7 @@ class Chapel < Formula
 
   depends_on "cmake"
   depends_on "gmp"
-  depends_on "llvm@15"
+  depends_on "llvm"
   depends_on "python@3.11"
```

Comment:
Can it work with 3.12?

Reply:
I think it can (locally it worked, anyway), I'll remove that and the python assignment below it

Reply:
Some deprecated regex strings (probably need to be raw strings):
```
  /opt/homebrew/Cellar/chapel/2.0/libexec/util/chplenv/compiler_utils.py:35: SyntaxWarning: invalid escape sequence '\d'
    match = re.search(u'(\d+)(\.(\d+))?(\.(\d+))?(\.(\d+))?', version_string)
```

I guess the `checkChplInstall` script expects no output.

Reply:
Thanks @cho-m! 
The change to python 3.12 did expose the need to update this string, however we are past our code freeze window and would like to request that we be allowed to stay at 3.11 for now, knowing that we will fix it for the next release in 3 months.

Reply:
Sure, just wanted to check if we could